source/geometries/Next100Vessel.cc

```diff
@@ -472,15 +547,17 @@ namespace nexus {
           G4ThreeVector glob_vtx(vertex);
           // this->GetCoordOrigin() only has x and y set
           glob_vtx = glob_vtx - GetCoordOrigin() - G4ThreeVector(0, 0, gate_z_pos_);
-          VertexVolume =
-            geom_navigator_->LocateGlobalPointAndSetup(glob_vtx, 0, false);
+          VertexVolume = geom_navigator_->LocateGlobalPointAndSetup(glob_vtx, 0, false);
         } while (VertexVolume->GetName() != "VESSEL");
       }
     }
 
     else if (region == "PORT_1a"){
-      vertex = port_gen_->GenerateVertex(VOLUME);
-
+      if (th_source_ == "next100") {
+        vertex = th_port_gen_->GenerateVertex(VOLUME);
+      } else if (th_source_ == "next_white") {
+        vertex = th_white_port_gen_->GenerateVertex(VOLUME);
+      }
       vertex = vertex.rotateX( 90. * deg);
```

Comment:
If I understand correctly these lines, for each port, it defines a vertex based on the detector and then adapts it and places it. However, this makes some NEW ports to be placed very far from the active volume. Am I missing anything?

Reply:
This code is only used in NEXT100 (it is in the NEXT100Vessel class). I think that the misunderstanding comes from the `next_white` name of the source. As I mentioned in another comment, if you can see another way of identifying the two sources, please, let me know!